### PR TITLE
Mark Maven-exported dependencies as scope provided

### LINKED ISSUE #125: Mark Maven-exported dependencies as scope provided
To prevent unnecessary artifact downloads with Maven 3 all dependencies provided by the Maven distribution should have scope "provided" (and only be listed with the version shipping with the minimum Maven version being supported). For further info look at https://issues.apache.org/jira/browse/MPLUGIN-370 and https://issues.apache.org/jira/browse/MNG-7097.

### PR DESCRIPTION
**Pull request by kwin**
This closes #125

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by kwin** [comment]
@cziegeler Can you review?

File: parent-oss/pom.xml [hunk 1]
```diff
@@ -54,6 +54,7 @@ governing permissions and limitations under the License.
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <!-- this is used for both enforcing the minimum java version and as compile target -->
     <project.java.version>1.8</project.java.version>
+    <maven.plugin.version>3.6.2</maven.plugin.version>
```
Comment (kwin) [inline parent-oss/pom.xml:57]: We need to wait for 3.6.4 due to https://issues.apache.org/jira/browse/MPLUGIN-384

Comment (cziegeler) [reply parent-oss/pom.xml:57]: What is the relation of the maven.plugin version and the minimal maven version required for the resulting maven plugin? Or in other words if we use 3.6.2 of maven plugin plugin for the aemanalyser plugin, do we need to require 3.6.2 for the aemanalyser plugin?

Comment (cziegeler) [reply parent-oss/pom.xml:57]: That said , I wouldn't mind requiring 3.6.3

Comment (kwin) [reply parent-oss/pom.xml:57]: You have to distinguish between maven-plugin-api (a run time requirement has an impact on the minimum Maven version required) and Maven Plugin Tools (https://maven.apache.org/plugin-tools/index.html, only relevant at build time). This PR only upgrades the plugin build tools without an impact on the minimum Maven requirement when aemanalyser is used.

Comment (cziegeler) [reply parent-oss/pom.xml:57]: Sure, understood - but this plugin does something which ends up in the plugin. If you are sure that this has no maven version dependency, then ok

Comment (kwin) [reply parent-oss/pom.xml:57]: Yes, it has no impact on the run time. The plugin descriptor generated is still the same version as always: https://maven.apache.org/ref/3.8.4/maven-plugin-api/plugin.html

**Comment by kwin** [comment]
I now reverted the update to a newer m-plugin-p as this is not really related, also still waiting for the release of 3.6.4. This is now ready for merge.

File: aemanalyser-maven-plugin/pom.xml [hunk 2]
```diff
@@ -101,27 +93,34 @@ governing permissions and limitations under the License.
...
       <artifactId>maven-artifact</artifactId>
       <version>${maven.version}</version>
+      <scope>provided</scope>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.maven.plugin-tools</groupId>
+      <artifactId>maven-plugin-annotations</artifactId>
+      <version>3.6.1</version><!-- same version as m-plugin-p -->
+      <scope>provided</scope><!-- no run time retention -->
```
Comment (cziegeler) [inline aemanalyser-maven-plugin/pom.xml:121]: Plugin version from parent pom is 3.6.0 - maybe we should move the plugin definition from the parent pom into this one?

Comment (kwin) [reply aemanalyser-maven-plugin/pom.xml:121]: Would be good to manage both versions with the same property as version value in parent.

Comment (cziegeler) [reply aemanalyser-maven-plugin/pom.xml:121]: yes, either parent or this pom

Comment (kwin) [reply aemanalyser-maven-plugin/pom.xml:120]: Done in https://github.com/adobe/aemanalyser-maven-plugin/pull/126/commits/de6a6ade9c7daa0961812731caebee97111bc2e5

**Pull request merged**